Feature/gcg 44

papnoemi199999 opened the pull request:
- Moved the user search from the dashboard to the leaderboard page.
- Refactored the leaderboard’s styling to ensure consistent component sizing. 
- Extracted the language filter into a separate reusable component
- Added new filtering options: sort by ascending or descending order, and filter by a specific language so the leaderboard only shows users with XP in that language

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

refactor: extract leaderboard table in separate component
change: pagination style
refactor: move getLeaderboard logic from controller to service

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

refactor: move user search to leaderboard page

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

fix: reload only the leaderboard when paginating

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

add: descriptive text to the leaderboard for user engagement
fix: user search style

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

fix: magnifying glass icon to ensure proper layering
fix: adjusts styling of the "No buddies found" message

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

change: leaderboard query for enabling sorting by different criteria

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

add: asc/desc for leaderboard

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

refactor: language filter into a reusable LanguageFilter component

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

refactor: leaderboard component to include language-specific filtering functionality.

arnold-webgurus commented on app/Services/LeaderboardService.php:
```diff
@@ -0,0 +1,147 @@
+    private function getEmptyPaginator(): LengthAwarePaginator
+    {
+        return User::query()
+            ->whereNull('id')
```
The `id` field will never be empty.  You can delete this function, and use only the `buildLeaderboardQuery`

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

fix: reset pagination when filters change
fix: avatar display
remove: unnecessary empty pagination function, handle case when a language is selected but no user has that skill in buildLeaderboardQuery

papnoemi199999 committed:
GCG-44: As a team, I want the User Search moved into the Leaderboard so that discovery is faster.

fix: Dashboard imports and props
